### PR TITLE
pcb-bcr: Use the correct BCR register for Bay Trail CPUs

### PR DESCRIPTION
**Pull request by hughsie**
Fixes https://github.com/fwupd/fwupd/issues/2328

### CODE REVIEW

File: plugins/pci-bcr/fu-plugin-pci-bcr.c [hunk 1]
```diff
@@ -11,20 +11,45 @@
...
+{
+	FuPluginData *priv = fu_plugin_get_data (plugin);
+	guint32 eax = 0;
+	guint64 tmp;
+	g_autofree gchar *group = NULL;
+
+	/* has a quirk */
+	if (!fu_common_cpuid (0x1, &eax, NULL, NULL, NULL, error))
+		return FALSE;
+	group = g_strdup_printf ("DeviceInstanceId=CPUID\\PRO_%01X&FAM_%01X&MOD_%01X",
```
Comment (superm1) [inline plugins/pci-bcr/fu-plugin-pci-bcr.c:44]: As a another thought - shouldn't this populate from the cpu plugin?  Like CPU plugin can carry the quirk and some metadata w/ `BcrAddr`  And this plugin can lookup the device from the CPU plugin?

Comment (hughsie) [reply plugins/pci-bcr/fu-plugin-pci-bcr.c:44]: So we'd add the GUIDs to the `FuCpuDevice` and then set up a `fu_plugin_device_registered()` in the `pci-bcr` plugin, then check if the device has some kind of metadata using `fu_device_get_metadata_integer()` and if so set the `bcr_addr`?

Comment (superm1) [reply plugins/pci-bcr/fu-plugin-pci-bcr.c:44]: Exactly!

Comment (hughsie) [reply plugins/pci-bcr/fu-plugin-pci-bcr.c]: I've updated the patch with this suggestion; I agree it's a better design.

**Pull request merged**